core/authentication/auth_service_test.go

```diff
@@ -0,0 +1,65 @@
+package authentication_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/SpectoLabs/hoverfly/core/authentication"
+	. "github.com/onsi/gomega"
+)
+
+var timeLayout = "2006-01-02T15:04:05.000Z"
+
+func Test_HasReachFailedAttemptsLimit_ReturnsFalseIfAttemptsIsBelowOrEqualToLimit(t *testing.T) {
+	RegisterTestingT(t)
+
+	authentication.Attempts.Count = 0
+	Expect(authentication.HasReachedFailedAttemptsLimit(3, "10s")).To(BeFalse())
+
+	authentication.Attempts.Count = 1
+	Expect(authentication.HasReachedFailedAttemptsLimit(3, "10s")).To(BeFalse())
+
+	authentication.Attempts.Count = 2
+	Expect(authentication.HasReachedFailedAttemptsLimit(3, "10s")).To(BeFalse())
+
+	authentication.Attempts.Count = 3
+	Expect(authentication.HasReachedFailedAttemptsLimit(3, "10s")).To(BeFalse())
+}
+
+func Test_HasReachFailedAttemptsLimit_ReturnsTrueIfAttemptsIsAboveLimit(t *testing.T) {
+	RegisterTestingT(t)
+
+	authentication.Attempts.Count = 4
+	Expect(authentication.HasReachedFailedAttemptsLimit(3, "10s")).To(BeFalse())
```

Comment:
Ah yeah, that's a false positive. The reason it passes is because no date is set last failed, which would never happen in the real system.

I have updated the test.